frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Base64 } from 'js-base64';
+import store from '@console/internal/redux';
+import { LoadingBox } from '@console/internal/components/utils';
+import { connectToFlags, WithFlagsProps, FlagsObject } from '@console/internal/reducers/features';
+import { FLAGS } from '@console/shared';
+import { WSFactory } from '@console/internal/module/ws-factory';
+import { resourceURL } from '@console/internal/module/k8s';
+import { PodModel } from '@console/internal/models';
+import { Terminal } from './AutoResizeTerminal';
+
+// pod exec WS protocol is FD prefixed, base64 encoded data (sometimes json stringified)
+
+// Channel 0 is STDIN, 1 is STDOUT, 2 is STDERR (if TTY is not requested), and 3 is a special error channel - 4 is C&C
+// The server only reads from STDIN, writes to the other three.
+// see also: https://github.com/kubernetes/kubernetes/pull/13885
+
+type CloudShellExecProps = WithFlagsProps & {
+  container: string;
+  podname: string;
+  namespace: string;
+  shcommand?: string[];
+  message?: string;
+  flags?: FlagsObject;
+};
+
+type CloudShellExecState = {
+  open: boolean;
+  error: string;
+};
+
+const NO_SH =
+  'starting container process caused "exec: \\"sh\\": executable file not found in $PATH"';
+
+class CloudShellExec extends React.PureComponent<CloudShellExecProps, CloudShellExecState> {
+  private terminal;
+
+  private ws;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      error: null,
+    };
+    this.terminal = React.createRef();
+  }
+
+  connect = () => {
+    const { container, podname, namespace, shcommand } = this.props;
+    const usedClient = this.props.flags[FLAGS.OPENSHIFT] ? 'oc' : 'kubectl';
```

Comment:
In consistent access of `this.props`. Line above deconstructs all props.

Reply:
fixed